### PR TITLE
Refactor callback service and controller

### PR DESCRIPTION
**Pull request by PaulDoyle-DEFRA**
This change refactors the Govpay payment callback route and service to better reflect the Govpay payment flow, as opposed to aligning closely with the Worldpay payment flow. It also removes a redundant second call to the Govpay API to query payment status.
https://eaflood.atlassian.net/browse/RUBY-1891
https://eaflood.atlassian.net/browse/RUBY-1892
https://eaflood.atlassian.net/browse/RUBY-1893
https://eaflood.atlassian.net/browse/RUBY-1894

### CODE REVIEW

**Review by kennyevil — DISMISSED** [review]
The controllers look to be as compact as they can be without becoming unreadable and the request specs are pretty thorough. LGTM.

**Review by Beckyrose200 — DISMISSED** [review]
From what I can tell looks good :D

**Review by d-a-v-e — DISMISSED** [review]
Amazing chunk of work! looks ready to go to QA 👍

File: config/locales/forms/renewal_received_pending_govpay_payment_forms/en.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+en:
+  waste_carriers_engine:
+    renewal_received_pending_govpay_payment_forms:
+      new:
+        title: Application received
```
[the remark this answers is not in the document]
Comment (PaulDoyle-DEFRA) [reply config/locales/forms/renewal_received_pending_govpay_payment_forms/en.yml:5]: As discussed: Team preferences is to not use quotes unless necessary, so I have updated the two locale files in the PR accordingly.

**Review by d-a-v-e — APPROVED** [review]

**Pull request merged**